lib/pipeline.js

```diff
@@ -24,7 +22,7 @@ class PipelineModel extends BaseModel {
      * @param  {Function} callback              fn(err, data) where data is the newly created object
      */
     create(config, callback) {
-        const id = hashr.sha1(config.scmUrl);
+        const id = this.generateId(config);
```

Comment:
This might not work. Config sometimes includes configUrl and sometimes not. I think you need to create a new object here that only includes scmUrl.